### PR TITLE
Add more integration tests.

### PR DESCRIPTION
**Pull request by alext**
This adds integration tests for creating/editing topics.  It also adds pending assertions for sending both topics and mainstream_browse_pages to the publishing-api.

### CODE REVIEW

File: spec/features/mainstream_browse_pages_spec.rb [hunk 1]
```diff
@@ -55,6 +64,45 @@
...
+      "description" => "Voting",
+      "format" => "mainstream_browse_page",
+    })
+
+    # And the page should have been updated in Panopticon
+    assert_tag_updated_in_panopticon(
+      :tag_type => 'section',
+      :tag_id => 'citizenship',
+      :title => 'Citizenship in the UK',
+      :description => 'Voting',
```
Comment (rboulton) [inline spec/features/mainstream_browse_pages_spec.rb]: should this not send the fact that this is in draft state to panopticon?

Comment (alext) [inline spec/features/mainstream_browse_pages_spec.rb]: It should, but that's not what the code currently does.  These have therefore revealed a bug.  I'll pick that up in a separate PR if that's ok...

File: spec/features/topic_create_edit_spec.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    end
+
+    # And a draft should have been sent to publishing-api (pending)
+    #assert_publishing_api_put_draft_item('/working-at-sea', {
+      #"title" => "Working at sea",
+      #"description" => 'The sea, the sky, the sea, the sky...',
+      #"format" => "topic",
+    #})
+
+    # And the topic should have been created in Panopticon
```
Comment (rboulton) [inline spec/features/topic_create_edit_spec.rb]: in draft state

**Comment by rboulton** [comment]
This looks good to me.  I'll merge, so you can look at the draft state bug separately.

File: spec/features/mainstream_browse_pages_spec.rb (hunk 1 above)
Comment (alext) [inline spec/features/mainstream_browse_pages_spec.rb]: Actually, after further investigation, the Panopticon API has a separate `/publish` endpoint for changing the state of a tag from draft to published.  The create/update endpoints don't change the state.

**Pull request merged**